spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/MultipleJGitEnvironmentRepositoryFactory.java

```diff
@@ -29,23 +29,27 @@
 		MultipleJGitEnvironmentProperties> {
 	private ConfigurableEnvironment environment;
 	private ConfigServerProperties server;
-	private Optional<TransportConfigCallback> transportConfigCallback;
 
-	public MultipleJGitEnvironmentRepositoryFactory(ConfigurableEnvironment environment, ConfigServerProperties server,
-													Optional<TransportConfigCallback> transportConfigCallback) {
+	public MultipleJGitEnvironmentRepositoryFactory(ConfigurableEnvironment environment, ConfigServerProperties server) {
 		this.environment = environment;
 		this.server = server;
-		this.transportConfigCallback = transportConfigCallback;
 	}
 
 	@Override
 	public MultipleJGitEnvironmentRepository build(MultipleJGitEnvironmentProperties environmentProperties) {
 		MultipleJGitEnvironmentRepository repository = new MultipleJGitEnvironmentRepository(environment,
 				environmentProperties);
-		repository.setTransportConfigCallback(transportConfigCallback.orElse(null));
+		repository.setTransportConfigCallback(propertiesBasedSshTransportCallback(environmentProperties));
```

Comment:
I guess it's better to leave `transportConfigCallback` for custom configuration and check whether it's provided or not here.